bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -152,63 +157,70 @@ public static String executeUrl(String httpMethod, String url, Properties httpHe
+        request.onComplete(listener);
+
         if (httpHeaders != null) {
             for (String httpHeaderKey : httpHeaders.stringPropertyNames()) {
-                method.addRequestHeader(new Header(httpHeaderKey, httpHeaders.getProperty(httpHeaderKey)));
+                request.header(httpHeaderKey, httpHeaders.getProperty(httpHeaderKey));
             }
         }
         // add content if a valid method is given ...
-        if (method instanceof EntityEnclosingMethod && content != null) {
-            EntityEnclosingMethod eeMethod = (EntityEnclosingMethod) method;
-            eeMethod.setRequestEntity(new InputStreamRequestEntity(content, contentType));
+        if (method.equals(HttpMethod.POST) || method.equals(HttpMethod.PUT) && content != null) {
+            request.content(new InputStreamContentProvider(content), contentType);
         }
 
-        Credentials credentials = extractCredentials(url);
+        Map<String, String> credentials = extractCredentials(url);
         if (credentials != null) {
-            client.getParams().setAuthenticationPreemptive(true);
-            client.getState().setCredentials(AuthScope.ANY, credentials);
+            Authentication.Result result = createBasicAuthenticationResult(url, 0, credentials.get(USERNAME_KEY),
+                    credentials.get(PASSWORD_KEY), false);
+            client.getAuthenticationStore().addAuthenticationResult(result);
         }
 
-        Logger logger = LoggerFactory.getLogger(HttpUtil.class);
         if (logger.isDebugEnabled()) {
-            try {
-                logger.debug("About to execute '" + method.getURI().toString() + "'");
-            } catch (URIException e) {
-                logger.debug(e.getMessage());
-            }
+            logger.debug("About to execute {}", request.getURI());
         }
 
         try {
-
-            int statusCode = client.executeMethod(method);
-            if (statusCode >= HttpStatus.SC_BAD_REQUEST) {
-                logger.debug("Method failed: " + method.getStatusLine());
+            ContentResponse response = request.send();
+            int statusCode = response.getStatus();
+            if (statusCode >= HttpStatus.BAD_REQUEST_400) {
+                String statusLine = statusCode + " " + response.getReason();
+                logger.debug("Method failed: {}", statusLine);
             }
 
-            String responseBody = IOUtils.toString(method.getResponseBodyAsStream());
+            String responseBody = response.getContentAsString();
             if (!responseBody.isEmpty()) {
                 logger.trace(responseBody);
             }
 
             return responseBody;
-        } catch (HttpException he) {
-            logger.error("Fatal protocol violation: {}", he.toString());
-        } catch (IOException ioe) {
-            logger.error("Fatal transport error: {}", ioe.toString());
+        } catch (Exception e) {
+            logger.error("Fatal transport error: {}", e);
```

Comment:
This clutters the log with long exceptions, if a host is not available. That's why originally, only the message was logged, not the exception.